core/src/proof.rs

```diff
@@ -100,6 +100,45 @@ pub struct VerifiedPathProof {
     terminal: Option<LeafData>,
 }
 
+impl VerifiedPathProof {
+    /// Get the terminal node.
```

Comment:
Feels weird to say that it gets you the terminal node while returning None.

Reply:
Fair. I added more docs but will punt on adding an `enum TerminalNode`